From 902402fcd69a8208766568c898fd719b43c9eeeb Mon Sep 17 00:00:00 2001
From: Azhar Shaikh <azhar.shaikh@intel.com>
Date: Thu, 19 Nov 2020 11:22:49 -0700
Subject: [PATCH] Add a flag to build 'vendor'

Make compiling vendor specific code optional.
By default set it to 'no'.

Signed-off-by: Azhar Shaikh <azhar.shaikh@intel.com>
---
 Makefile.am  | 7 ++++++-
 configure.ac | 9 ++++++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index d28175a..4eea2b1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,7 +24,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = common decode encode vainfo videoprocess vendor/intel vendor/intel/sfcsample
+SUBDIRS = common decode encode vainfo videoprocess
 
 if USE_X11
 SUBDIRS += putsurface
@@ -38,6 +38,11 @@ if ENABLE_TESTS
 SUBDIRS += test
 endif
 
+if ENABLE_VENDOR_INTEL
+SUBDIRS += vendor/intel
+SUBDIRS += vendor/intel/sfcsample
+endif
+
 # Extra clean files so that maintainer-clean removes *everything*
 MAINTAINERCLEANFILES = \
 	aclocal.m4 compile config.guess config.sub \
diff --git a/configure.ac b/configure.ac
index 7a4d97f..13249fa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -94,6 +94,11 @@ AC_ARG_ENABLE([tests],
     [], [enable_tests="no"])
 
 
+AC_ARG_ENABLE([vendor_intel],
+    [AC_HELP_STRING([--enable-vendor-intel],
+                    [build intel specific code @<:@default=no@:>@])],
+    [], [enable_vendor_intel="no"])
+
 LT_INIT
 AC_DISABLE_STATIC
 AC_PROG_CC
@@ -215,6 +220,7 @@ if test "$USE_DRM:$USE_X11:$USE_WAYLAND" = "no:no:no"; then
 fi
 
 AM_CONDITIONAL(ENABLE_TESTS, test "$enable_tests" = "yes")
+AM_CONDITIONAL(ENABLE_VENDOR_INTEL, test "$enable_vendor_intel" = "yes")
 
 AC_OUTPUT([
     Makefile
@@ -242,5 +248,6 @@ echo
 echo Libva VA-API version ............. : $LIBVA_API_VERSION
 echo Installation prefix .............. : $prefix
 echo Extra window systems ............. : $BACKENDS
-echo Enable Unit-tests .................... : $enable_tests
+echo Enable Unit-tests ................ : $enable_tests
+echo Enable vendor-Intel .............. : $enable_vendor_intel
 echo
-- 
2.17.1
